src/Controls/src/Core/Handlers/Items2/iOS/TemplatedCell2.cs

```diff
@@ -107,49 +108,42 @@ public override void PrepareForReuse()
 
 		public void Bind(DataTemplate template, object bindingContext, ItemsView itemsView)
 		{
-			if (PlatformHandler is null)
-			{
-				var virtualView = template.CreateContent(bindingContext, itemsView) as View;
-
-				var mauiContext = itemsView.FindMauiContext()!;
-				var nativeView = virtualView!.ToPlatform(mauiContext);
-
-				PlatformView = nativeView;
-
-				PlatformHandler = virtualView.Handler as IPlatformViewHandler;
-
-				InitializeContentConstraints(nativeView);
-
-				virtualView.BindingContext = bindingContext;
-			}
-
-			if (PlatformHandler?.VirtualView is View view)
-			{
-				view.SetValueFromRenderer(BindableObject.BindingContextProperty, bindingContext);
-			}
+			var virtualView = template.CreateContent(bindingContext, itemsView) as View;
+			BindVirtualView(virtualView, bindingContext, itemsView, true);
 		}
 
 		public void Bind(View virtualView, ItemsView itemsView)
+		{
+			BindVirtualView(virtualView, itemsView.BindingContext, itemsView, false);
```

Comment:
Is this `itemsView.BindingContext` correct? Will that be bindng an item to the whole VS and not the item object?

Reply:
I suppose that is what the existing code was doing, but just checking to see if the OG code was correct.

Reply:
Yeah this is for the Headers, so basically if you don t give a template and just give a plain headerView or footerview. I just refactored the code here